api/src/main/java/nl/fhict/happynews/api/auth/oauth2/MongoTokenStore.java

```diff
@@ -0,0 +1,118 @@
+package nl.fhict.happynews.api.auth.oauth2;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.oauth2.common.OAuth2AccessToken;
+import org.springframework.security.oauth2.common.OAuth2RefreshToken;
+import org.springframework.security.oauth2.provider.OAuth2Authentication;
+import org.springframework.security.oauth2.provider.token.AuthenticationKeyGenerator;
+import org.springframework.security.oauth2.provider.token.DefaultAuthenticationKeyGenerator;
+import org.springframework.security.oauth2.provider.token.TokenStore;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+@Component
+public class MongoTokenStore implements TokenStore {
+
+    private final MongoAccessTokenRepository accessTokenRepository;
+
+    private final MongoRefreshTokenRepository refreshTokenRepository;
+
+    private AuthenticationKeyGenerator authenticationKeyGenerator = new DefaultAuthenticationKeyGenerator();
+
+    @Autowired
+    public MongoTokenStore(final MongoAccessTokenRepository accessTokenRepository,
+                           final MongoRefreshTokenRepository refreshTokenRepository) {
+        this.accessTokenRepository = accessTokenRepository;
+        this.refreshTokenRepository = refreshTokenRepository;
+    }
+
+    @Override
+    public OAuth2Authentication readAuthentication(OAuth2AccessToken token) {
+        return readAuthentication(token.getValue());
+    }
+
+    @Override
+    public OAuth2Authentication readAuthentication(String tokenId) {
+        return accessTokenRepository.findByTokenId(tokenId).getAuthentication();
+    }
+
+    @Override
+    public void storeAccessToken(OAuth2AccessToken token, OAuth2Authentication authentication) {
+        AuthenticationAccessToken accessToken = new AuthenticationAccessToken(token,
+            authentication, authenticationKeyGenerator.extractKey(authentication));
+        accessTokenRepository.save(accessToken);
+    }
+
+    @Override
+    public OAuth2AccessToken readAccessToken(String tokenValue) {
+        AuthenticationAccessToken token = accessTokenRepository.findByTokenId(tokenValue);
+        if (token == null) {
```

Comment:
Why not throw an exception here, when you **do** do that in MongoUserDetailsService?